src/main/java/org/opensearch/neuralsearch/query/NeuralSparseQueryBuilder.java

```diff
@@ -94,7 +101,11 @@ public NeuralSparseQueryBuilder(StreamInput in) throws IOException {
     protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeString(fieldName);
         out.writeString(queryText);
-        out.writeString(modelId);
+        if (isClusterOnOrAfterMinReqVersionForDefaultModelIdSupport()) {
+            out.writeOptionalString(this.modelId);
+        } else {
+            out.writeString(this.modelId);
```

Comment:
If the cluster version smaller than min version, shouldn't here be `writeOptionalString` since the modelId is null?

Reply:
If the cluster version is before min version (i.e. 2.13), the model id field is required, it's not optional. So it can not be null and we can use writeString here